ethers-providers/src/transports/http.rs

```diff
@@ -69,10 +70,19 @@ impl JsonRpcClient for Provider {
         params: T,
     ) -> Result<R, ClientError> {
         let next_id = self.id.fetch_add(1, Ordering::SeqCst);
-
         let payload = Request::new(next_id, method, params);
 
-        let res = self.client.post(self.url.as_ref()).json(&payload).send().await?;
+        let mut client = self.client.post(self.url.as_ref()).json(&payload);
+        if let Some(auth) = &self.authorization {
+            client = match auth {
+                Authorization::Basic(username, password) => {
+                    client.basic_auth(username, Some(password))
+                }
+                Authorization::Bearer(token) => client.bearer_auth(token),
+            };
+        }
```

Comment:
instead of doing this, we should rather set the default headers for the client
https://docs.rs/reqwest/latest/reqwest/blocking/struct.ClientBuilder.html#method.default_headers